client/ayon_blender/plugins/publish/validate_no_animation.py

```diff
@@ -0,0 +1,64 @@
+
+import bpy
+import inspect
+from ayon_core.pipeline.publish import (
+    ValidateContentsOrder,
+    OptionalPyblishPluginMixin,
+    PublishValidationError
+)
+from ayon_blender.api.action import (
+    SelectInvalidAction,
+    RepairAction
+)
+from ayon_blender.api import plugin
+
+
+class ValidateNoAnimation(
+    plugin.BlenderInstancePlugin,
+    OptionalPyblishPluginMixin
+):
+    """Ensure that meshes do not have animation data."""
+
+    order = ValidateContentsOrder
+    hosts = ["blender"]
+    families = ["blendScene", "model", "rig"]
+    label = "No Animation"
+    actions = [SelectInvalidAction, RepairAction]
+
+    @classmethod
+    def get_invalid(cls, instance):
+        invalid = []
+        for obj in instance:
+            if isinstance(obj, bpy.types.Object):
+                if obj.animation_data and obj.animation_data.action:
+                    invalid.append(obj)
+        return invalid
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            self.log.debug("Skipping Validate No Animation...")
+            return
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            names = ", ".join(obj.name for obj in invalid)
+            raise PublishValidationError(
+                "Objects found in instance which have"
+                f" animation data: {names}",
+                title="Keyframes on Objects",
+                description=self.get_description()
+            )
+
+    def get_description(self):
+        return inspect.cleandoc("""
+            ### Keyframes on Objects
+
+            Objects must not contain any keyframe animation data.
+            Please remove all keyframes before publishing.
+        """)
+
+    @classmethod
+    def repair(cls, instance):
+        invalid_objects = cls.get_invalid(instance)
+        for obj in invalid_objects:
+            obj.animation_data.action = None
```

Comment:
Out of curiosity - in Blender, if you do this. Does it actually _delete_ the action/animation datablock if after this it has no users of that datablock anymore. Or does it remain in the scene data?

Reply:
It will still last in datablock, it is just removed in the action editor. You can always re-link it back if you want to.
![image](https://github.com/user-attachments/assets/b62f71a8-6cc9-481b-b7de-9cd954a983d8)